buffer/test/com/treode/buffer/PagedBufferSpec.scala

```diff
@@ -48,33 +48,65 @@ class PagedBufferSpec extends FlatSpec {
   capacity (64, 2)
   capacity (65, 3)
 
-  def discard (nbytes: Int, npages: Int) {
-    it should (s"discard the right pages for nbytes=$nbytes") in {
+  def discard (nbytes: Int, rbytes: Int, wbytes: Int, npages: Int) {
+    it should (s"discard the right pages for nbytes=$nbytes, rbytes=$rbytes, wbytes=$wbytes") in {
       val buffer = PagedBuffer (pageBits)
-      buffer.writePos  = (4 << pageBits) - 1
-      buffer.readPos = nbytes
+      buffer.writePos = wbytes
+      buffer.readPos = rbytes
+      val initpages = buffer.capacity >> pageBits;
+
       val before = (Seq (buffer.pages: _*), buffer.writePos, buffer.readPos)
-      buffer.discard (nbytes)
-      assertResult ((4 - npages) << pageBits) (buffer.capacity)
+      val discarded = buffer.discard (nbytes)
       val after = (Seq (buffer.pages: _*), buffer.writePos, buffer.readPos)
-      for (i <- 0 until before._1.length - npages)
-        assert (before._1 (i + npages) == after._1 (i))
-      for (i <- before._1.length - npages until after._1.length)
-        assert (after._1 (i) == null)
-      assertResult (before._2 - pageSize * npages) (after._2)
-      assertResult (before._3 - pageSize * npages) (after._3)
-    }}
+
+      //Calculating the number of bytes discarded manually
+      val expectedDiscard = (if (nbytes == wbytes) (nbytes) else (npages * pageSize))
+      
+      //Checking if the right number of bytes are discarded
+      assertResult (discarded) (expectedDiscard)  
```

Comment:
I think you have switched expected and actual. The signature for `assertResult` is

    def assertResult (expected) (actual)